Common/include/QuasarThreadPool.h

```diff
@@ -0,0 +1,69 @@
+/* © Copyright CERN, 2018.  All rights not expressly granted are reserved.
+ * QuasarThreadPool.h
+ *
+ *  Created on: 4 May 2018
+ *      Author: Piotr Nikiel <piotr@nikiel.info>
+ *
+ *  This file is part of Quasar.
+ *
+ *  Quasar is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public Licence as published by
+ *  the Free Software Foundation, either version 3 of the Licence.
+ *
+ *  Quasar is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public Licence for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Quasar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COMMON_INCLUDE_QUASARTHREADPOOL_H_
+#define COMMON_INCLUDE_QUASARTHREADPOOL_H_
+
+#include <mutex>
+#include <vector>
+#include <thread>
+#include <list>
+#include <condition_variable>
+
+#include <statuscode.h>
+
+namespace Quasar
+{
+
+class ThreadPoolJob
+{
+public:
+    virtual ~ThreadPoolJob() {};
+
+    virtual void execute() = 0;
+};
+
+class ThreadPool
+{
+public:
+    ThreadPool (unsigned int maxThreads, unsigned int maxJobs);
+    ~ThreadPool ();
+
+    UaStatus addJob (ThreadPoolJob* job);
+
+private:
+    void work();
+
+    std::mutex m_accessLock;
+    bool m_quit;
+    std::vector<std::thread> m_workers;
+    std::list<ThreadPoolJob*> m_pendingJobs;
```

Comment:
or maybe a std::queue

Reply:
Maybe ;-) But I don't see a strong technical advantage here. I will consider it though.

Reply:
As you wish: it is a cosmetic/subjective point. I mentioned it because I thought queue seemed a clearer indication of the intent of the code than list

Reply:
OK I bought your idea.